dfusion_rust_core/src/models/order.rs

```diff
@@ -32,9 +38,40 @@ impl Serializable for u128 {
     }
 }
 
+impl From<Arc<Log>> for Order {
+    fn from(log: Arc<Log>) -> Self {
+        let mut bytes: Vec<u8> = log.data.0.clone();
+        Order {
+            slot: U256::pop_from_log_data(&mut bytes),
+            slot_index: u16::pop_from_log_data(&mut bytes),
+            account_id: u16::pop_from_log_data(&mut bytes),
+            sell_token: u8::pop_from_log_data(&mut bytes),
```

Comment:
buyToken comes before sellToken in the event data. Thus it needs to be popped first.